docs/DevEnv.md

```diff
@@ -0,0 +1,66 @@
+## Netplugin Developer's Guide
+This document describes the development environment and typical steps to hack on netplugin.
+While there are alternative ways, the steps listed here is used by a lot of developers while
+making incremental changes and doing frequent compilation for a quicker development cycle.
+
+Happy hacking!
+
+### 1. Check-out a tree. `Estimated time: 2 minutes`
+Notes: Make sure GOPATH is set. This is a one time activity.
+```
+$ cd $GOPATH
+$ mkdir -p src/github.com/contiv
+$ cd src/github.com/contiv
+# it is recommended that you fork the repo if you want to make contributions
+$ git clone https://github.com/<your-github-id>/netplugin.git
+
+# you can clone contiv repo, if you like playing with some code before forking it
```

Comment:
since this is a development guide I would avoid adding steps for non developers here i.e. users who are not planning to planning to make code changes. We might want to have a separate demo/trying out guide for that.